src/ccnl-core/src/ccnl-malloc.c

```diff
@@ -33,27 +34,50 @@ void*
 debug_malloc(size_t s, const char *fn, int lno, char *tstamp)
 #endif
 {
-    struct mhdr *h = (struct mhdr *) malloc(s + sizeof(struct mhdr));
+    size_t size;
+    /** check if the operation can be performed without causing an integer overflow */
+    if (!INT_ADD_OVERFLOW(s, sizeof(struct mhdr), size)) {
+        struct mhdr *h = (struct mhdr *) malloc(size); 
+        /** memory allocation failed */
+        if (!h) {
+            return NULL;
+        }
+
+        h->next = mem;
+        mem = h;
+        h->fname = (char *) fn;
+        h->lineno = lno;
+        h->size = s;
 
-    if (!h)
-        return NULL;
-    h->next = mem;
-    mem = h;
-    h->fname = (char *) fn;
-    h->lineno = lno;
-    h->size = s;
 #ifdef CCNL_ARDUINO
-    h->tstamp = tstamp;
+        h->tstamp = tstamp;
 #else
-    //h->tstamp = strdup(tstamp);
-    h->tstamp = strcpy( malloc( strlen(tstamp)+1), tstamp );
-#endif
-    /*
-    if (s == 32) CONSOLE("+++ s=%d %p at %s:%d\n", s,
-                         (void*)(((unsigned char *)h) + sizeof(struct mhdr)),
-                         (char*) fn, lno);
-    */
-    return ((unsigned char *)h) + sizeof(struct mhdr);
+        size_t new_timestamp_size;
+        /** determine size of the timestamp */
+        size_t timestamp_size = strlen(tstamp);
+        /** check if +1 can safely be added */
+        if (!INT_ADD_OVERFLOW(timestamp_size, 1, new_timestamp_size)) {
```

Comment:
I get an error here: 
error: use of undeclared identifier 'new_timestamp_size'.

Same error in line 60

Reply:
I fixed that 2 minutes ago.